libuavcan_drivers/stm32/driver/include/uavcan_stm32/thread.hpp

```diff
@@ -17,6 +17,12 @@
 # include <ctime>
 # include <cstring>
 #elif UAVCAN_STM32_BAREMETAL
+#elif UAVCAN_STM32_FREERTOS
+# include <stm32f4xx.h>
+# include <cmsis_os.h>
+  #ifndef MAX_SEMAPHORE_COUNT
+    #define MAX_SEMAPHORE_COUNT      50
```

Comment:
This is the only place where the semaphore count can be decremented: https://github.com/larics/libuavcan/blob/master/libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp#L874. As you can see, it is not guaranteed to be reachable. Therefore, the semaphore may overflow. I suggest to use a different synchronization primitive, probably an idempotent one, like a conditional variable or whatever FreeRTOS has to offer nowadays.